plugin/pkg/scheduler/algorithm/priorities/node_affinity_test.go

```diff
@@ -32,62 +32,72 @@ func TestNodeAffinityPriority(t *testing.T) {
 	label4 := map[string]string{"abc": "az11", "def": "az22"}
 	label5 := map[string]string{"foo": "bar", "key": "value", "az": "az1"}
 
-	affinity1 := map[string]string{
-		v1.AffinityAnnotationKey: `
-		{"nodeAffinity": {"preferredDuringSchedulingIgnoredDuringExecution": [
-			{
-				"weight": 2,
-				"preference": {
-					"matchExpressions": [
-						{
-							"key": "foo",
-							"operator": "In", "values": ["bar"]
-						}
-					]
-				}
-			}
-		]}}`,
+	affinity1 := &v1.Affinity{
+		NodeAffinity: &v1.NodeAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{{
+				Weight: 2,
+				Preference: v1.NodeSelectorTerm{
+					MatchExpressions: []v1.NodeSelectorRequirement{{
+						Key:      "foo",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"bar"},
+					}},
+				},
+			}},
+		},
 	}
 
-	affinity2 := map[string]string{
-		v1.AffinityAnnotationKey: `
-		{"nodeAffinity": {"preferredDuringSchedulingIgnoredDuringExecution": [
-			{
-				"weight": 2,
-				"preference": {"matchExpressions": [
-					{
-						"key": "foo",
-						"operator": "In", "values": ["bar"]
-					}
-				]}
-			},
-			{
-				"weight": 4,
-				"preference": {"matchExpressions": [
-					{
-						"key": "key",
-						"operator": "In", "values": ["value"]
-					}
-				]}
-			},
-			{
-				"weight": 5,
-				"preference": {"matchExpressions": [
-					{
-						"key": "foo",
-						"operator": "In", "values": ["bar"]
+	affinity2 := &v1.Affinity{
+		NodeAffinity: &v1.NodeAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{
+				{
+					Weight: 2,
+					Preference: v1.NodeSelectorTerm{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "foo",
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"bar"},
+							},
+						},
 					},
-					{
-						"key": "key",
-						"operator": "In", "values": ["value"]
+				},
+				{
+					Weight: 4,
+					Preference: v1.NodeSelectorTerm{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "key",
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"value"},
```

Comment:
Is there an example where this is actually a string array vs. just a string?  Just wondering if we should rationalize the field types.